src/pydvl/influence/inversion.py

```diff
@@ -53,51 +52,135 @@ def solve_hvp(
+
+    @classmethod
+    def register(
+        cls,
+        model_type: Type[TwiceDifferentiable],
+        inversion_method: InversionMethod,
+        overwrite: bool = False,
+    ):
+        """
+        Register a function for a specific model type and inversion method.
+
+        The function to be registered must conform to the following signature:
+        `(model: TwiceDifferentiable, training_data: DataLoaderType, b: TensorType, hessian_perturbation: float = 0.0, ...)`.
+
+        :param model_type: The type of the model the function should be registered for.
+        :param inversion_method: The inversion method the function should be registered for.
+        :param overwrite: If True, allows overwriting of an existing registered function for the same model type and inversion method.
+                      If False, logs a warning when attempting to register a function for an already registered model type and inversion method.
+
+
+        :raises TypeError: If the provided model_type or inversion_method are of the wrong type.
+        :raises ValueError: If the function to be registered does not match the required signature.
+        :return: A decorator for registering a function.
+        """
+
+        if not isinstance(model_type, type):
+            raise TypeError(f"'model_type' must be a Type[TwiceDifferentiable]")
+
+        if not isinstance(inversion_method, InversionMethod):
+            raise TypeError(f"'inversion_method' must be an InversionMethod")
+
+        key = (model_type, inversion_method)
+
+        def decorator(func):
+            if not overwrite and key in cls.registry:
+                warnings.warn(
+                    f"There is already a function registered for model type {model_type} and inversion method {inversion_method}. "
+                    f"To overwrite the existing function {cls.registry.get(key)} with {func}, set overwrite to True."
+                )
+            sig = inspect.signature(func)
+            params = list(sig.parameters.values())
+
+            expected_args = [
+                ("model", model_type),
+                ("training_data", DataLoaderType.__bound__),
+                ("b", model_type.tensor_type()),
+                ("hessian_perturbation", float),
+            ]
+
+            for (name, typ), param in zip(expected_args, params):
+                if not (
+                    isinstance(param.annotation, typ)
+                    or issubclass(param.annotation, typ)
+                ):
+                    raise ValueError(
+                        f'Parameter "{name}" must be of type "{typ.__name__}"'
+                    )
+
+            @functools.wraps(func)
+            def wrapper(*args, **kwargs):
+                return func(*args, **kwargs)
+
+            cls.registry[key] = wrapper
+            return wrapper
+
+        return decorator
+
+    @classmethod
+    def get_registered(
```

Comment:
The suffix "registered" seems redundant. Consider its use: `InversionRegistry.get_registered(...)` vs `InversionRegistry.get(...)`. As a rule of thumb, I think we should prefer conciseness if it makes code read out aloud in a natural way.

```suggestion
    def get(
```